src/axom/sina/core/Datum.hpp

```diff
@@ -0,0 +1,206 @@
+ *   myArrayDatum.setTags(tags);
+ *   myRecord->add(myArrayDatum);
+ * \endcode
+ */
+class Datum
+{
+public:
+  /**
+     * \brief Construct a new Datum.
+     *
+     * \param value the string value of the datum
+     */
+  Datum(const std::string &value);
+
+  /**
+     * \brief Construct a new Datum.
+     *
+     * \param value the double value of the datum
+     */
+  Datum(const double &value);
+
+  /**
+     * \brief Construct a new Datum.
+     *
+     * \param value the string array value of the datum
+     */
+  Datum(const std::vector<std::string> &value);
+
+  /**
+     * \brief Construct a new Datum.
+     *
+     * \param value the scalar array value of the datum
+     */
+  Datum(const std::vector<double> &value);
+
+  /**
+     * \brief Construct a Datum from its Node representation.
+     *
+     * \param asNode the Datum as conduit Node
+     */
+  explicit Datum(conduit::Node const &asNode);
+
+  /**
+     * \brief Get the string value of the Datum.
+     *
+     * \return the string value
+     */
+  std::string const &getValue() const noexcept { return stringValue; }
+
+  /**
+     * \brief Get the scalar value of the Datum.
+     *
+     * \return the scalar value
+     */
+  double const &getScalar() const noexcept { return scalarValue; }
+
+  /**
+     * \brief Get the string array value of the Datum.
+     *
+     * \return the string vector value
+     */
+  std::vector<std::string> const &getStringArray() const noexcept
+  {
+    return stringArrayValue;
+  }
+
+  /**
+     * \brief Get the scalar array value of the Datum.
+     *
+     * \return the scalar vector value
+     */
+  std::vector<double> const &getScalarArray() const noexcept
+  {
+    return scalarArrayValue;
+  }
+
+  /**
+     * \brief Get the tags of the Datum
+     *
+     * \return the tags of the value
+     */
+  std::vector<std::string> const &getTags() const noexcept { return tags; }
+
+  /**
+     * \brief Set the tags of the Datum
+     *
+     * \param tags the tags of the value
+     */
+  void setTags(const std::vector<std::string> &tags);
+
+  /**
+     * \brief Get the units of the Datum
+     *
+     * \return the units of the value
+     */
+  std::string const &getUnits() const noexcept { return units; }
+
+  /**
+     * \brief Set the units of the Datum
+     *
+     * \param units the units of the value
+     */
+  void setUnits(const std::string &units);
+
+  /**
+     * \brief Get the type of the Datum
+     *
+     * \return the type of the value
+     */
+  ValueType getType() const noexcept { return type; }
+
+  /**
+     * \brief Convert this Datum to its conduit Node representation.
+     *
+     * \return the Node representation of this Datum.
+     */
+  conduit::Node toNode() const;
+
+private:
+  std::string stringValue;
+  double scalarValue;
+  std::vector<std::string> stringArrayValue;
+  std::vector<double> scalarArrayValue;
+  std::string units;
+  std::vector<std::string> tags;
```

Comment:
Observation: It looks like several classes in Sina have optional units and/or tags.  Perhaps this is something that could abstracted in the future (if there is sufficient shared logic), e.g. into a private base class.